### ISSUE TITLE
`PIVOT` mistakenly parsed as an `alias_expression` after a subquery

### ISSUE THREAD

**Issue by michael-the1** [sqlfluff/sqlfluff#2877]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing a query that contains a subquery in the FROM statement, followed by a PIVOT, sqlfluff recognizes the PIVOT as an alias.

Example query:
```sql
SELECT *
FROM (SELECT * FROM my_tbl)
PIVOT (min(f_val) FOR f_id IN (1, 2)) AS f (a, b);
```

### Expected Behaviour

Parse `PIVOT` as a `from_pivot_expression`

### Observed Behaviour

Parses `PIVOT` as an `alias_expression`.

Using the following minimal example:

```bash
echo "SELECT * FROM (SELECT * FROM my_tbl) PIVOT (min(f_val) FOR f_id IN (1, 2)) AS f (a, b);" | sqlfluff parse --dialect=snowflake -
```

I get the following error

```
==== parsing violations ====
L:   1 | P:  44 |  PRS | Line 1, Position 44: Found unparsable section: '(min(f_val) FOR
                       | f_id IN (1, 2)) AS f (a,...'
```

Though, really, the main offender is in the parse tree.

```
[L:  1, P: 38]      |                        alias_expression:                                                                                                                                                    
[L:  1, P: 38]      |                            identifier:                       'PIVOT' 
```

### How to reproduce

```bash
echo "SELECT * FROM (SELECT * FROM my_tbl) PIVOT (min(f_val) FOR f_id IN (1, 2)) AS f (a, b);" | sqlfluff parse --dialect=snowflake -
```

### Dialect

snowflake

### Version

0.11.1

### Configuration
```
[sqlfluff]
dialect = snowflake
; sqlfluff rules: https://docs.sqlfluff.com/en/stable/rules.html
rules = L001,
        ; Avoid trailing whitespace
        L002,
        ; Avoid mixing tabs and spaces
        L005,
        ; Avoid space before a comma
        L006,
        ; Use a single space around an operator
        L007,
        ; Use a newline before an operator, not after
        L008,
        ; Use a single space after a comma
        L009,
        ; Use a trailing newline at end of file
        L010,
        ; Use capital letters for keywords
        L015,
        ; Avoid parentheses after DISTINCT
        L020,
        ; Use unique aliases within a clause
        L021,
        ; Avoid ambiguous DISTINCT with GROUP BY
        L023,
        ; Use a single space after AS in a WITH clause
        L040,
        ; Use uppercase for TRUE, FALSE, NULL
        L041,
        ; Use DISTINCT and other SELECT modifiers on the same line as the clause
        L043,
        ; Avoid when/then/else statements returning boolean values
        L047,
        ; Use COUNT(*), not COUNT(0) or COUNT(1)
        Picnic_L001,
        ; Use consistent aliasing styles
        Picnic_L002,
        ; Use explicit aliases
        Picnic_L003,
        ; Use a single space around a casting operator
        Picnic_L004,
        ; Do not read from DV deprecated tables
        Picnic_L005,
        ; Avoid pattern IFF(x, TRUE, FALSE)
```

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L040]
capitalisation_policy = upper

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Changing `FromExpressionElementSegment` seems to fix this and allow it to parse if you fancy putting in a PR?

```
@@ -1154,7 +1154,7 @@ class FromExpressionElementSegment(BaseSegment):
         OptionallyBracketed(Ref("TableExpressionSegment")),
         OneOf(
             Ref("AliasExpressionSegment"),
-            exclude=Ref("SamplingExpressionSegment"),
+            exclude=OneOf(Ref("SamplingExpressionSegment"),Ref("JoinLikeClauseGrammar")),
             optional=True,
         ),
```

**Comment by michael-the1** [comment]
Yes, I'll put in a PR 💪

**Comment by PLBMR** [comment]
Sorry @michael-the1 ; I beat you to the PR. The change motivated by #2877 was relevant to addressing #2875, so I fixed both in #2888 .
